src/sonic-yang-models/yang-models/sonic-extension.yang

```diff
@@ -20,4 +24,45 @@ module sonic-extension {
         description "Key regex used to convert config DB keys to YANG Config";
         argument "value";
     }
+
+    extension db-name {
+        description "DB name, e.g. APPL_DB, CONFIG_DB";
+        argument "value";
+    }
```

Comment:
provide more justification on the needness of extension

Reply:
Updated SONiC YANG Guidelines with details of each extension.
Raised PR: https://github.com/Azure/SONiC/pull/754